v2/tools/generator/internal/codegen/storage/resource_conversion_graph_test.go

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package storage
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/test"
+)
+
+func TestResourceConversionGraph_WithSingleReference_HasExpectedTransition(t *testing.T) {
+	/*
+	 * Test that a graph that contains just one API version and the associate Storage version ends up with a single
+	 * transition, between the original API version and the storage variant.
+	 */
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	person2020 := astmodel.MakeTypeName(test.Pkg2020, "Person")
+	person2020s := astmodel.MakeTypeName(test.Pkg2020s, "Person")
+
+	builder := NewResourceConversionGraphBuilder("demo", "v")
+	builder.Add(person2020, person2020s)
+	graph, err := builder.Build()
+
+	// Check size of graph
+	g.Expect(err).To(Succeed())
+	g.Expect(graph.TransitionCount()).To(Equal(1))
+
+	// Check for single expected transition
+	next, ok := graph.LookupTransition(person2020)
+	g.Expect(ok).To(BeTrue())
+	g.Expect(next).NotTo(BeNil())
+	g.Expect(astmodel.IsStoragePackageReference(next.PackageReference)).To(BeTrue())
+}
+
+func TestResourceConversionGraph_WithTwoGAReferences_HasExpectedTransitions(t *testing.T) {
+	/*
+	 * Test that a graph that contains two GA API releases and the matching storage versions ends up with three
+	 * transitions. Each API version should have a transition to its matching storage variant, and there should be a
+	 * transition from the older storage variant to the newer one.
+	 */
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	person2020 := astmodel.MakeTypeName(test.Pkg2020, "Person")
+	person2020s := astmodel.MakeTypeName(test.Pkg2020s, "Person")
+	person2021 := astmodel.MakeTypeName(test.Pkg2021, "Person")
+	person2021s := astmodel.MakeTypeName(test.Pkg2021s, "Person")
+
+	builder := NewResourceConversionGraphBuilder("demo", "v")
+	builder.Add(person2020, person2020s)
+	builder.Add(person2021, person2021s)
+	graph, err := builder.Build()
+
+	// Check size of graph
+	g.Expect(err).To(Succeed())
+	g.Expect(graph.TransitionCount()).To(Equal(3))
+
+	// Check for expected transition for Person2020
+	after2020, ok := graph.LookupTransition(person2020)
+	g.Expect(ok).To(BeTrue())
+	g.Expect(after2020).NotTo(BeNil())
+
+	// Check for expected transition for Person2021
+	after2021, ok := graph.LookupTransition(person2021)
+	g.Expect(ok).To(BeTrue())
+	g.Expect(after2021).NotTo(BeNil())
+
+	// Check for expected transition for the storage variant of Person2020s
+	after2020s, ok := graph.LookupTransition(after2020)
+	g.Expect(ok).To(BeTrue())
+	g.Expect(after2020s).To(Equal(after2021))
+}
+
+func TestResourceConversionGraph_WithGAAndPreviewReferences_HasExpectedTransitions(t *testing.T) {
```

Comment:
Do these two tests together cover the case I ran into where we have:

`compatv1previewstorage -> v1apistorage -> v1apipreviewstorage`?

Or is that not actually really possible anymore because the `v1apistorage` node didn't contain the `FederatedIdentityCredential` type so in actuality now the issue I ran into boils down to a simple:
`compatv1previewstorage -> v1apipreviewstorage`?

Though I guess you could still test that we're correctly "skipping" the `v1apistorage` step there for this resource type -- would that be in `conversion_graph_test.go` and do we have such a test (I'm not seeing it but possibly that's because it just doesn't look how I expected).